jwcrypto/jwe.py

```diff
@@ -89,6 +90,7 @@ def __init__(self, plaintext=None, protected=None, unprotected=None,
         self._allowed_algs = None
         self.objects = dict()
         self.plaintext = None
+        self.header_registry = JWSEHeaderRegistry(DefaultJWEHeaderRegistry)
```

Comment:
Why the rewrap here ?

Reply:
Or maybe why DefaultJWEHeaderRegistry is a JWSEHeaderRegistry() itself instead of being just a dict ?

Reply:
One reason I thought about this was in the current use, the header functions would have to be registered separately on each instantiation. I was thinking in the case where we had in mind to add functionality for a particular header throughout the program, in that case, we would add the header to the `DefaultRegistry`, and every object `JWS/JWE` created from then on would have that functionality built it.

Reply:
uhmm, changing the DefaultJWEHeaderRegistry would make everything brittle, because i would change how every caller works, including callers in other libraries.
Please do not do that.

By allowing to pass a new registry via an init parameter in the JWE you can create your own registry once, and then always pass that one into your own controlled JWEs.

This allow the "cost savings" you want without messing up static global objects.

so I think let's get rid of DefaultJWEHeaderRegistry, and maintain only the dictionary, and move initialization in JWE/JWS constructors, including the ability to use a passed in Registry.

Reply:
Yea - that makes sense. I will rename it back to `JWEHeaderRegistry` as well then so it doesn't break any libraries referencing that particular variable.